src/meta/src/hummock/compaction_scheduler.rs

```diff
@@ -88,26 +106,30 @@ where
                     break 'compaction_trigger;
                 }
             };
-            self.pick_and_assign(compaction_group).await;
+            request_channel.unschedule(compaction_group);
```

Comment:
Shall we unschedule after get_compact_task? Looks like all compact request happening before get_compact_task will be considered altogether.

Reply:
Do you mean to avoid `unschedule` if `get_compact_task` returns `None`?
It is by intention because we have proactively scheduled another request if previous `pick_and_assign` succeeds, while this new request may not `get_compact_task` and should not longer be scheduled.

Reply:
I get that but that is not what I mean. 

What I mean is should we invoke `unschedule` after `get_compact_task` instead of before it? If we invoke `unschedule` before `get_compact_task`, all compact requests happening between `unschedule` and `get_compact_task` will be scheduled while technically these requests should already be considered in the current run of `get_compact_task`.

Reply:
fixed